plugins/modules/net_tools/pritunl/pritunl_user.py

```diff
@@ -0,0 +1,344 @@
+        **dict_merge(
+            get_pritunl_settings(module),
+            {"filters": {"name": org_name}},
+        )
+    )
+
+    if len(org_obj_list) == 0:
+        module.fail_json(
+            msg="Can not add user to organization '%s' which does not exist" % org_name
+        )
+
+    org_id = org_obj_list[0]["id"]
+
+    # Grab existing users from this org
+    users = list_pritunl_users(
+        **dict_merge(
+            get_pritunl_settings(module),
+            {
+                "organization_id": org_id,
+                "filters": {"name": user_name},
+            },
+        )
+    )
+
+    # Check if the pritunl user already exists
+    if len(users) > 0:
+        # Compare remote user params with local user_params and trigger update if needed
+        user_params_changed = False
+        for key in user_params.keys():
+            # When a param is not specified grab existing ones to prevent from changing it with the PUT request
+            if user_params[key] is None:
+                user_params[key] = users[0][key]
+
+            # 'groups' is a list comparison
+            if key == "groups":
+                if set(users[0][key]) != set(user_params[key]):
+                    user_params_changed = True
+
+            # otherwise it is either a boolean or a string
+            else:
+                if users[0][key] != user_params[key]:
+                    user_params_changed = True
+
+        # Trigger a PUT on the API to update the current user if settings have changed
+        if user_params_changed:
+            response = post_pritunl_user(
+                **dict_merge(
+                    get_pritunl_settings(module),
+                    {
+                        "organization_id": org_id,
+                        "user_id": users[0]["id"],
+                        "user_data": user_params,
+                    },
+                )
+            )
+
+            result["changed"] = True
+            result["response"] = response
+        else:
+            result["changed"] = False
+            result["response"] = users
+    else:
+        response = post_pritunl_user(
+            **dict_merge(
+                get_pritunl_settings(module),
+                {
+                    "organization_id": org_id,
+                    "user_data": user_params,
+                },
+            )
+        )
+        result["changed"] = True
+        result["response"] = response
+
+    module.exit_json(**result)
+
+
+def remove_pritunl_user(module):
+    result = {}
+
+    org_name = module.params.get("organization")
+    user_name = module.params.get("user_name")
+
+    org_obj_list = []
+    try:
+        org_obj_list = list_pritunl_organizations(
+            **dict_merge(
+                get_pritunl_settings(module),
+                {
+                    "filters": {"name": org_name},
+                },
+            )
+        )
+    except PritunlException:
+        pass
```

Comment:
This is still not a good way of handling the issue, IMO. The root cause of the problem will be discarded here and the user will receive the "Can not remove user ..." message below. The gap between the error message the user sees and the real problem that occurred will only make it harder to debug.

Reply:
Personally I'd remove this try/except here entirely. Let the exception cut through all the layers and the exact error to be show to the user.

Reply:
Alternatively, use `from ansible.module_utils.six import raise_from` and then `raise_from(NewException(), old_exception)`. If you return the exception in the `fail_json` call with `exception=traceback.format_exc()`, you can see both exceptions (no idea how it looks with Python 2, but with Python 3 it should work well).